src/main/java/Containers/Portfolio.java

```diff
@@ -23,67 +23,116 @@ public static Portfolio getInstance() {
+        this.assetList = new ArrayList<>();
+        this.transactionList = new ArrayList<>();
+        this.votingHistory = new ArrayList<>();
+        this.profitability = 0.0;
+    }
+    // Getter and Setter for profitability
+    public double getProfitability() {
+        return profitability;
+    }
+
+    public void setProfitability(double profitability) {
+        this.profitability = profitability;
+    }
+
+    // Getter and Setter for assets, transactions, and votes
+
+    public List<Asset> getAssetList() {
+        return assetList;
+    }
+
+    public List<Transaction> getTransactionList() {
+        return transactionList;
+    }
+
+    public List<Vote> getVotingHistory() {
+        return votingHistory;
     }
 
     // Add an asset to the system.
     // This method only takes a snapshot of its parameter, the parameter object can be safely modified afterwards.
     public void add(Asset asset0) {
-        Asset asset = this.storage.get(asset0.getType());
-        if(asset == null) {
-            asset = new Asset(0, asset0.getPrice(), asset0.getType(), asset0.getSymbol());
-            this.storage.put(asset.getType(), asset);
-        }
-        asset.setVolume(asset.getVolume() + asset0.getVolume());
+        this.assetList.add(asset0);
+    }
+
+    // Add an asset to the system.
+    public void add(Transaction transaction0) {
+        this.transactionList.add(transaction0);
+    }
+
+    // Add a vote to the system.
+    public void add(Vote vote0) {
+        this.votingHistory.add(vote0);
     }
 
     // Subtracts an asset from the system.
     // This method only takes a snapshot of its parameter, the parameter object can be safely modified afterwards.
     public void sub(Asset asset0) {
-        Asset asset = this.storage.get(asset0.getType());
-        if(asset == null) {
-            asset = new Asset(0, asset0.getPrice(), asset0.getType(), asset0.getSymbol());
-            this.storage.put(asset.getType(), asset);
+        if (this.assetList.contains(asset0)) {
+            this.assetList.remove(asset0);
+        }
+        else{
+            System.out.println("Current portfolio does not contain " + asset0);
+        }
+    }
+
+    public void sub(Transaction transaction0) {
+        if (this.transactionList.contains(transaction0)) {
+            this.transactionList.remove(transaction0);
+        }
+        else{
+            System.out.println("Current portfolio does not contain " + transaction0);
+        }
+    }
+
+    public void sub(Vote vote0) {
+        if (this.votingHistory.contains(vote0)) {
+            this.votingHistory.remove(vote0);
+        }
+        else{
+            System.out.println("Current portfolio does not contain " + vote0);
         }
-        asset.setVolume(asset.getVolume() - asset0.getVolume());
     }
 
     // Calculates the value of a specific asset.
     // If the asset is not found, this method will return zero.
-    public double getValue(String type) {
```

Comment:
I think this method is unnecessary. The value of the asset can be accessed directly from the Asset instance, so all the Portfolio has to do is to search for that Asset and return it.